config/config_partial.go

```diff
@@ -118,27 +126,65 @@ func DecodeBaseBlocks(ctx *ParsingContext, file *hclparse.File, includeFromChild
 		evalParsingContext,
 	)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	trackInclude, err := getTrackInclude(ctx, terragruntIncludeList, includeFromChild)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+
+	// set feature flags
+	tgFlags := terragruntFeatureFlags{}
+	// load default feature flags
+	if err := file.Decode(&tgFlags, evalParsingContext); err != nil {
+		return nil, nil, nil, err
+	}
+
+	evaluatedFlags := map[string]cty.Value{}
+	// copy default feature flags to evaluated flags
+
+	for name, value := range ctx.TerragruntOptions.FeatureFlags {
+		contextFlag, err := flagToCtyValue(name, value)
+
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		evaluatedFlags[name] = contextFlag
+	}
+
+	for _, flag := range tgFlags.FeatureFlags {
+		if _, exists := evaluatedFlags[flag.Name]; !exists {
+			contextFlag, err := flagToCtyValue(flag.Name, *flag.Default)
```

Comment:
Does it make more sense to call this `evaluatedFlag` here? I'm not sure why it's called `contextFlag`.

Reply:
I still don't get why this is `contextFlag`, but it's not a big deal.